src/backend/distributed/commands/create_distributed_table.c

```diff
@@ -236,7 +239,20 @@ master_create_distributed_table(PG_FUNCTION_ARGS)
 		for (attributeIndex = 0; attributeIndex < attributeCount; attributeIndex++)
 		{
 			AttrNumber attributeNumber = attributeNumberArray[attributeIndex];
-			if (distributionColumn->varattno == attributeNumber)
+
+			if (distributionColumn->varattno != attributeNumber)
+			{
+				continue;
+			}
+
+			if (indexInfo->ii_Unique)
+			{
+				hasDistributionColumn = true;
+				break;
+			}
+
+			if (indexInfo->ii_ExclusionOps != NULL &&
+				OperatorImplementsEquality(indexInfo->ii_ExclusionOps[attributeIndex]))
```

Comment:
Instead of having two `if` statements with the same body, could we merge them with some helper variables to something like this:

```
bool uniqueConstraint = indexInfo->ii_Unique;
bool exclusionConstraintWithEquality = indexInfo->ii_ExclusionOps != NULL && OperatorImplementsEquality(indexInfo->ii_ExclusionOps[attributeIndex]);

if (uniqueConstraint || exclusionConstraintWithEquality)
{
    hasDistributionColumn = true;
    break;
}

```